File: data-structures/linked-lists/array-representation/array-representation.c
#include <stdio.h>
#include <stdlib.h>


#define NONE -1


int
main(void)
{
    int new = NONE, start = NONE, ptr = NONE;
    int *info = NULL, *link = NULL;
    int list_size, i;

    printf("Enter list size: ");
    scanf("%d", &list_size);

    for(i = 1; i <= list_size; i++) {
        new++;
        info = realloc(info, sizeof(*info) * (new + 1));
        link = realloc(link, sizeof(new) * (new + 1));

        if(start == NONE) {
            start = new, ptr = new;
        } else {
            link[ptr] = new;
            ptr = link[ptr];
        }

        printf("Enter item %d: ", i);
        scanf("%d", &info[ptr]);

        link[ptr] = NONE;
    }

    printf("\nList items: ");
    while(start != NONE) {
        printf("%d, ", info[start]);
        start = link[start];
    }

    return 0;
}

File: data-structures/linked-lists/array-representation/array-representation.py
info = []
link = []

start = None
ptr = None
new = -1


# List creation
list_size = int(input("Enter list size: "))

for i in range(1, list_size + 1):
    new += 1

    if start == None:
        start = new
        ptr = start

    else :
        link[ptr] = new
        ptr = link[ptr]

    item = input(f"Enter item {i}: ")
    info.append(item)
    link.append(None)

#List traversal

print("\nList items are: ")

while start != None:
    print(f"{info[start]}", end=', ')
    start = link[start]